### ISSUE TITLE
Writing documentation

### ISSUE THREAD

**Issue by Namnodorel** [felar/robug#4]
- [x] How does a Neural Network work in general? _(What problems can they solve, how are they structured, how does the math work)_
- [x] Optimization Algorithms (Overview of which ones exist and what they are usually used for)
- [x] How does one improve the performance of a NN? _(dropout, ĺearning rate decay, etc.)_
- [x] What kinds of neural networks are there? (general overview of the types and kind of stuff they are used for)
- [x] How does reinforcement learning work? _(Why is it needed in contrast to a normal NN, how is it different, how does the math work?)_
- [x] Advanced Reinforcement Learning overview (Deep Q Learning, and similar)
- [x] Tensorflow Agents Concepts
- [x] What our bot has learned so far
- [x] Ideas how the bot could learn better in the future
